[CI] IndexAuditUpgradeIT.testAuditLogs fails

## https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+bwc-tests/8/console

## Reproduction line
does not reproduce locally
```
 ./gradlew :x-pack:qa:rolling-upgrade:with-system-key:v6.3.1#upgradedClusterTestRunner \
  -Dtests.seed=57D936B12E792F96 \
  -Dtests.class=org.elasticsearch.upgrades.IndexAuditUpgradeIT \
  -Dtests.method="testAuditLogs" \
  -Dtests.security.manager=true \
  -Dtests.locale=bg \
  -Dtests.timezone=Pacific/Marquesas
```

## Example relevant log:
```
00:00:46    > Throwable #1: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2343, key=upgraded-node-0}, {doc_count=2175, key=node-1}, {doc_count=2106, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 	at __randomizedtesting.SeedInfo.seed([57D936B12E792F96:300A75E395A08B0D]:0)
00:00:46    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 	at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 	at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:830)
00:00:46    > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:804)
00:00:46    > 	at org.elasticsearch.upgrades.IndexAuditUpgradeIT.testAuditLogs(IndexAuditUpgradeIT.java:68)
00:00:46    > 	at java.lang.Thread.run(Thread.java:748)
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46   2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+6.4+bwc-tests/x-pack/qa/rolling-upgrade/with-system-key/build/testrun/v6.3.1#upgradedClusterTestRunner/J0/temp/org.elasticsearch.upgrades.IndexAuditUpgradeIT_57D936B12E792F96-001
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46   2> NOTE: test params are: codec=Asserting(Lucene70): {}, docValues:{}, maxPointsInLeafNode=174, maxMBSortInHeap=7.47073894872366, sim=RandomSimilarity(queryNorm=true): {}, locale=bg, timezone=Pacific/Marquesas
00:00:46   2> NOTE: Linux 4.4.0-1061-aws amd64/Oracle Corporation 1.8.0_181 (64-bit)/cpus=16,threads=1,free=433534592,total=514850816
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46   2> NOTE: All tests run in this JVM: [IndexingIT, RollupIDUpgradeIT, TokenBackwardsCompatibilityIT, IndexAuditUpgradeIT]
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=1983, key=upgraded-node-0}, {doc_count=1794, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=2025, key=upgraded-node-0}, {doc_count=1841, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2175, key=node-1}, {doc_count=2141, key=upgraded-node-0}, {doc_count=1933, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46    > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=2431, key=node-2}, {doc_count=2212, key=upgraded-node-0}, {doc_count=2175, key=node-1}, {doc_count=2000, key=upgraded-node-1}, {doc_count=891, key=node-0}]
00:00:46    > Expected: a collection with size <6>
00:00:46    >      but: collection size was <5>
00:00:46    > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
00:00:46    > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
00:00:46    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:818)
00:00:46    > 		... 39 more
00:00:46 Completed [4/6] in 10.40s, 1 test, 1 failure <<< FAILURES!
```

## Frequency

It happened 8 times in August, not super frequent but seems consistent.

Another failure with the same assertion 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+multijob-unix-compatibility/os=opensuse/33/console

```
   > Throwable #1: java.lang.AssertionError: Found node buckets [{doc_count=630, key=upgraded-node-1}]
   > Expected: a collection with size <0>
   >      but: collection size was <1>
   > 	at __randomizedtesting.SeedInfo.seed([81138282C946761F:E6C0C1D0729FD284]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
   > 	at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:829)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:803)
   > 	at org.elasticsearch.upgrades.IndexAuditUpgradeIT.testAuditLogs(IndexAuditUpgradeIT.java:68)
   > 	at java.lang.Thread.run(Thread.java:748)
   > 	Suppressed: java.lang.AssertionError: Found node buckets [{doc_count=83, key=upgraded-node-1}]
   > Expected: a collection with size <0>
   >      but: collection size was <1>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.assertNumUniqueNodeNameBuckets(IndexAuditUpgradeIT.java:126)
   > 		at org.elasticsearch.upgrades.IndexAuditUpgradeIT.lambda$testAuditLogs$0(IndexAuditUpgradeIT.java:70)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:817)
   > 		... 39 more
...
```
Logs are attached

[IndexAuditUpgradeIT-Logs.zip](https://github.com/elastic/elasticsearch/files/2542345/IndexAuditUpgradeIT-Logs.zip)

There is no log for node `upgradedClusterTestCluster node0` as the test failed in the 2 thirds upgraded stage


```
./gradlew :x-pack:qa:rolling-upgrade:without-system-key:v5.6.13-SNAPSHOT#twoThirdsUpgradedTestRunner \
  -Dtests.seed=81138282C946761F \
  -Dtests.class=org.elasticsearch.upgrades.IndexAuditUpgradeIT \
  -Dtests.method="testAuditLogs" \
  -Dtests.security.manager=true \
  -Dtests.locale=pt-PT \
  -Dtests.timezone=GB-Eire
```

--- a/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/index/IndexAuditTrail.java
+++ b/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/index/IndexAuditTrail.java
@@ -277,7 +277,7 @@ public class IndexAuditTrail implements AuditTrail, ClusterStateListener {
         }
 
         if (TemplateUtils.checkTemplateExistsAndVersionMatches(INDEX_TEMPLATE_NAME, SECURITY_VERSION_STRING,
-                clusterState, logger, Version.CURRENT::onOrAfter) == false) {
+                clusterState, logger, Version.CURRENT::onOrBefore) == false) {
             logger.debug("security audit index template [{}] is not up to date", INDEX_TEMPLATE_NAME);
             return false;
         }
@@ -306,6 +306,15 @@ public class IndexAuditTrail implements AuditTrail, ClusterStateListener {
             index = resolve(IndexAuditTrailField.INDEX_NAME_PREFIX, first.timestamp, rollover);
         }
         return index;
+    }
+
+    private boolean hasStaleMessage() {
+        final Message first = peek();
+        if (first == null) {
+            return false;
+        }
+        return false == IndexNameResolver.resolve(first.timestamp, rollover)
+                .equals(IndexNameResolver.resolve(DateTime.now(DateTimeZone.UTC), rollover));
     }
 
     /**
@@ -381,7 +390,7 @@ public class IndexAuditTrail implements AuditTrail, ClusterStateListener {
             IndexMetaData indexMetaData = indices.get(0);
             MappingMetaData docMapping = indexMetaData.mapping("doc");
             if (docMapping == null) {
-                if (indexToRemoteCluster || state.nodes().isLocalNodeElectedMaster()) {
+                if (indexToRemoteCluster || state.nodes().isLocalNodeElectedMaster() || hasStaleMessage()) {
                     putAuditIndexMappingsAndStart(index);
                 } else {
                     logger.trace("audit index [{}] is missing mapping for type [{}]", index, DOC_TYPE);
@@ -399,7 +408,7 @@ public class IndexAuditTrail implements AuditTrail, ClusterStateListener {
                 if (versionString != null && Version.fromString(versionString).onOrAfter(Version.CURRENT)) {
                     innerStart();
                 } else {
-                    if (indexToRemoteCluster || state.nodes().isLocalNodeElectedMaster()) {
+                    if (indexToRemoteCluster || state.nodes().isLocalNodeElectedMaster() || hasStaleMessage()) {
                         putAuditIndexMappingsAndStart(index);
                     } else if (versionString == null) {
                         logger.trace("audit index [{}] mapping is missing meta field [{}]", index, SECURITY_VERSION_STRING);

